source/Planet.cpp

```diff
@@ -231,6 +233,28 @@ void Planet::Load(const DataNode &node)
 
 
 
+void Planet::FinishLoading(Set<Wormhole> &wormholes)
+{
+	// If this planet is in multiple systems, then it is a wormhole.
+	if(!wormhole && systems.size() > 1)
+	{
+		wormhole = wormholes.Get(name);
+		wormhole->LoadFromPlanet(this);
```

Comment:
That part will only get executed once, since after the first execution `this->wormhole != nullptr`. Calling LoadFromPlanet multiple times is safe yes.